### DISCUSSION TITLE
Is there a configuration parameter for a global max execution time for a query?

### DISCUSSION THREAD

**Discussion by AndreaWesterinen** [eclipse-rdf4j/rdf4j#3733] [Q&A]
I know that the timeout can be set at the per-query level, and that AbstractOperation holds a maxExecutionTime default (0).

But, is there a way to override the default via a configuration parameter for a repository?

And, does the maxExecutionTime override a per-query setting? For example, if the max time is 60 seconds, but an individual query indicates that the timeout should be 10 mins.

Thanks.

Andrea

**Comment by abrokenjester** [comment]
I had to look this up, but no, there is no general setting for this.
